runtime/src/beta.rs

```diff
@@ -0,0 +1,171 @@
+use std::{
+    collections::BTreeMap,
+    convert::Infallible,
+    iter::FromIterator,
+    net::{Ipv4Addr, SocketAddr},
+    process::exit,
+};
+
+use anyhow::Context;
+use hyper::{
+    service::{make_service_fn, service_fn},
+    Body, Response, Server,
+};
+use shuttle_api_client::ShuttleApiClient;
+use shuttle_common::{
+    resource::{ResourceInput, ResourceState, Type},
+    secrets::Secret,
+};
+use shuttle_service::{ResourceFactory, Service};
+
+use crate::__internals::{Loader, Runner};
+
+const HEALTH_CHECK_PORT: u16 = 8001;
+
+pub async fn start(loader: impl Loader + Send + 'static, runner: impl Runner + Send + 'static) {
+    // Uses primitive parsing instead of clap for reduced dependency weight
+    let shuttle = std::env::var("SHUTTLE").is_ok();
+    let project_id = std::env::var("SHUTTLE_PROJECT_ID").expect("project id env var");
+    let project_name = std::env::var("SHUTTLE_PROJECT_NAME").expect("project name env var");
+    let env = std::env::var("SHUTTLE_ENV")
+        .expect("shuttle environment env var")
+        .parse()
+        .expect("invalid shuttle environment");
+    let ip = std::env::var("SHUTTLE_RUNTIME_IP")
+        .expect("runtime ip env var")
+        .parse()
+        .expect("invalid ip");
+    let port = std::env::var("SHUTTLE_RUNTIME_PORT")
+        .expect("runtime port env var")
+        .parse()
+        .expect("invalid port");
+    let api_url = std::env::var("SHUTTLE_API").expect("api env var");
```

Comment:
nit: Maybe we could still use a Config struct in a separate file, that is created by getting all these env vars? To clean up the function a bit.